### PR TITLE
Branch list

### PR DESCRIPTION
**Pull request by weedz**

### CODE REVIEW

File: src/Components/BranchList/index.tsx [hunk 1]
```diff
@@ -42,40 +42,48 @@ function BranchTree(branches: any) {
     )
 }
 
-export default class BranchList extends Component<any, {branches: ReturnType<typeof getBranchTree>, head?: BranchObj}> {
+type Props = {
+    branches?: BranchesObj
```
Comment (meion) [inline src/Components/BranchList/index.tsx:46]: Couldnt this be an array of branches instead? The obj-key and the Type seems mismatched.

File: src/Data/Actions.ts [hunk 2]
```diff
@@ -87,6 +87,8 @@ export type CommitObj = {
 
 export type BranchObj = {
     name: string
+    normalizedName: string
+    headSha: string
```
Comment (meion) [inline src/Data/Actions.ts:91]: Shouldn't "Sha" be capitialized?

File: src/Data/Provider.ts [hunk 3]
```diff
@@ -31,36 +31,47 @@ export async function getCommits(repo: Repository, start?: Commit, num: number =
...
+        refs.map(async (ref) => {
+            const headCommit = await ref.peel(Object.TYPE.COMMIT);
+            const refObj: BranchObj = {
+                name: ref.name(),
+                headSha: headCommit.id().tostrS(),
+                normalizedName: ""
+            };
+
+            if (ref.isBranch()) {
+                refObj.normalizedName = refObj.name.substring(11);
```
Comment (meion) [inline src/Data/Provider.ts:52]: magic numbers?

File: src/Data/Renderer/store.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    if (key) {
+        keyListeners[key].push(cb);
+    } else {
+        listeners.push(cb);
+    }
+}
+
+export function unsubscribe(cb: Function, key?: keyof StoreType) {
+    if (key) {
+        keyListeners[key].splice(listeners.indexOf(cb)>>>0, 1);
```
Comment (meion) [inline src/Data/Renderer/store.ts:37]: Comment these numbers, or put them in a constant so it's clear what this is

File: src/Components/BranchList/index.tsx (hunk 1 above)
Comment (weedz) [reply src/Components/BranchList/index.tsx:46]: It's an object with arrays of BranchObj. The type `BranchesObj` could be renamed to something better representing the actual type. Don't think that's an issue that should be resolved in this PR though.

**Comment by meion** [comment]
We should change namin of types in the soon future.

**Pull request merged**